lld/test/wasm/Inputs/signature-mismatch-debug-info-b.ll

```diff
@@ -0,0 +1,31 @@
+target triple = "wasm32-unknown-emscripten"
+
+define void @foo(i32 %a, i32 %b) !dbg !6 {
```

Comment:
Why does this not result in a link error due to multiple definitions of `foo`?

Reply:
I don't know. They are not marked as `weak`, so it's weird... I tested with system (x86) clang and gcc and they all seem to error out.

When we do this by LTO, meaning running `wasm-ld` directly on bitcode, they error out saying duplicate definition. When does this "signature mismatch" occur usually? When a definition and a callsite do not match in the signature?

Reply:
IIUC, regardless of signature... you can't have two global definitions of foo in the same program unless one of them is weak.  So I can't see why this wouldn't fail, even with matching signatures.